utils/client_helper.go

```diff
@@ -145,6 +153,28 @@ func getNodeInternalIP(node corev1.Node) (string, error) {
 	return "", fmt.Errorf("failed to find node internal ip address")
 }
 
+func getMachineNetwork(ctx context.Context, client runtimeclient.Client, nodeIp string) (string, error) {
+	installConfig, err := getInstallConfig(ctx, client)
+	if err != nil {
+		return "", fmt.Errorf("failed to get install config: %w", err)
+	}
+
+	for _, mn := range installConfig.Networking.MachineNetwork {
+		if _, _, err := net.ParseCIDR(mn.CIDR); err != nil {
+			return "", fmt.Errorf("MachineNetwork has an invalid CIDR: %s", mn.CIDR)
+		}
+		if _, ipnet, err := net.ParseCIDR(mn.CIDR); err == nil && ipnet.Contains(net.ParseIP(nodeIp)) {
+			return mn.CIDR, nil
+		}
+	}
+
+	// retuning empty string if no match found in order to be backward compatible as there is a possibility that
+	// that we upgrade cluster that was previously upgraded with lc-agent version that didn't set new machineNetwork
```

Comment:
Add logging, with the reason we return empty (node IP doesn't match?, no machine network?, machine network is invalid...)